pkg/distributor/distributor.go

```diff
@@ -1246,6 +1264,18 @@ func (d *Distributor) prePushValidationMiddleware(next PushFunc) PushFunc {
 			validatedExemplars += len(ts.Exemplars)
 		}
 
+		// Emit tracing span events for metrics with deduped samples.
+		spanLogger, ctx := spanlogger.NewWithLogger(ctx, d.log, "Distributor.prePushValidationMiddleware")
+		defer spanLogger.Finish()
+		for m, c := range dedupedPerMetric {
+			spanLogger.DebugLog(
+				"msg", "deduplicated samples/histograms with conflicting timestamps from write request",
+				"metric", m,
+				"count", c,
+				"tenant", userID,
```

Comment:
I don't think you need the tenant here, that's already inferred from the span.

Reply:
Done, thx.